src/streams.rs

```diff
@@ -91,10 +91,13 @@ impl BaseStream {
 
     #[cfg(feature = "tls")]
     fn connect_tls(host: Host<&str>, port: u16, info: &ConnectInfo) -> Result<BaseStream> {
-        let connector = TlsConnector::builder()
-            .danger_accept_invalid_certs(info.base_settings.accept_invalid_certs)
-            .danger_accept_invalid_hostnames(info.base_settings.accept_invalid_hostnames)
-            .build()?;
+        let mut connector_builder = TlsConnector::builder();
+        connector_builder.danger_accept_invalid_certs(info.base_settings.accept_invalid_certs);
+        connector_builder.danger_accept_invalid_hostnames(info.base_settings.accept_invalid_hostnames);
+        for cert in info.base_settings.root_certificates.iter() {
```

Comment:
I think the `.iter()` call could be replaced by just doing `for cert in &info.base_settings.root_certificates`?

Reply:
The compiler doesn't seems to like it:

```
error[E0507]: cannot move out of `info.base_settings.root_certificates` which is behind a shared reference
  --> src/streams.rs:97:21
   |
97 |         for cert in info.base_settings.root_certificates {
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |                     |
   |                     move occurs because `info.base_settings.root_certificates` has type `std::vec::Vec<certificate::Certificate>`, which does not implement the `Copy` trait
   |                     help: consider iterating over a slice of the `Vec<_>`'s content: `&info.base_settings.root_certificates`
```

Reply:
You need the `&` sigil in front to indicate that you want a reference and move out of the struct field, i.e. `in &info.base_settings.root_certificates` instead of `in info.base_settings.root_certificates`.